Move application module out of the project module

amatiashov opened the pull request:
- Configure static files
- Created template settings

firegurafiku commented:
Hi!

Why do we have to put all that static files under our version control? To me, they're obviously related to `django.contrib.admin` application, shouldn't they be hosted elsewhere? Django admin seems to work even without having them added.

Another point, the project structure. You say, it's not okay to have application module located inside of the project module. I don't mind separating them, but I really dislike that `_app` suffix and would like to prevent it appearing as a part of database table names. Renaming `scifight` application to `scifight_app` would also result in unnecessary hard database migration (which must be included with your PR too!).

So, let us have the following file structure:
```
scifight
├── manage.py
├── scifight_proj
│   ├── __init__.py
│   ├── settings.py
│   ├── settings_secret.py (← Okay with this name?)
│   ├── urls.py
│   └── wsgi.py
└── scifight
    ├── migrations
    │   ├── 0001_initial.py
    │   ├── __init__.py
    ├── admin.py
    ├── apps.py
    ├── models.py
    ├── tests.py
    └── views.py
```

Could you kindly rebase your pull-request to reflect this change? Also, please create separate commits for logically unrelated changes: one for changing project structure, and another for static files (if they're necessary to be added).

firegurafiku commented on scifight/migrations/0001_initial.py:
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-# Generated by Django 1.9.7 on 2016-07-17 03:22
```
No strong need to include this file into a commit if only one thing changed is a timestamp in a comment.

firegurafiku commented:
@turboj55 

In general, you shouldn't close PRs on your own, unless you want to say "Hey, I do not want this PR to be merged anymore". You did the great job improving your patch, but I still have something more to say, please don't take this personally.

### Commit message

You should write more concise descriptions (with better spelling, though). How about the following text?

```
Move scifight application top-level

Django guidelines [1] do not recommend putting application modules into
the directory where project settings live, so:

* Rename 'scifight' directory to 'scifight_proj'.
* Move 'scifight_proj/scifight' to just 'scifight'.
* Fix application name in settings.py and manage.py files.

[1]: <put the prooflink here>

Signed-off-by: (...)
```

Please note that the `Signed-off-by` thing is really required, as it indicates "Yes, I do want to license the contribution under the terms of AGPL, and I'm okay with the fact it's a viral copyleft license".

### Gitignore

Maybe you should make your `.gitignore` change a separate commit. It's pretty okay to have a one-line description for it, like:
```
.gitignore: add 'scifight/settings_secret.py' and '*.dylib'

Signed-off-by: (...)
```
I don't think it's a good idea to ignore *only* `libmysqlclient.18.dylib`, because you may have more files like those in future, even that file may eventually become `libmysqlclient.19.dylib` and escape your ignore pattern.

### default_app_config

What exactly for is `default_app_config = 'scifight.apps.WebsiteConfig'`? Does it have to be put into `__init__.py` file, or maybe there is some other place for it? I'd really prefer leaving these stupid `__init__.py` files empty.

### Separate commits

Please, have avoid putting major file renames, moves and deletes to the same commit with significant semantic changes. Keep them separated for the great good. It's also a good idea to have "whitespace-only" commits (code indentation, beautification and so on) separated from commits changing real code.

In this pull-request I'd be happy to see your commits like those:

* Move scifight application top-level
* .gitignore: add 'scifight/settings_secret.py' and '*.dylib'
* Add setting_secret_example.py
* (Describe what you did to `models.py` and `admin.py`.)

Note that the last change may be the subject to *another* PR, if you feel like that.

### Default locale

Please, don't change language and timezone to anything but they default values. The primary language for project is English, so let's delay it's translation to Russian until it's more or less ready.

amatiashov committed:
Move scifight application top-level

* Move scifight application folder top-level
    The default when creating applications within the project,
    Django creates an application in the project root

* Creating a template file for the secret-settings.

* Wrapping all application models decorator admin.register
    It will be convenient to configure the admin panel.

* Edit application models.
    on Fight team3 and team4 can be absent.
    on FightStage: during the initial filling of the jury are no estimates.

amatiashov committed:
Added sorting applications of models in the administrative panel

    Used django-modeladmin-reorder package.
    Pitched model group.

firegurafiku commented:
@turboj55 

I decided to take care of your commits my myself, so I rebased them from scratch, with slight changes. You can easily find them in project history, they're revisions 04c564c8ad6b0175d9237d4d5239af4015143eec to da3d87eb04f56038e008921f18644bc62a4da90a. I used your name and email during commit amending, so there should be no authorship problem. Please, link your email to GitHub account so your name in history became clickable.

A couple of notes: 

### Migrations!
 
Please, create new database migration each time you modify data models, and attach generated migration file to your commit. It also may be a good idea to append `(migration!)` mark to the title of commit message.

### Nullable many-to-many fields

Does it really make sense to have `null=True` and `blank=True` on `ManyToManyField`? Doesn't many-to-many imply 0-to-0 relations?
```diff
@@ -100,4 +100,4 @@ class FightStage(models.Model):
     refusals    = models.ManyToManyField(Problem, related_name="refusals")
-    points      = models.ManyToManyField(JuryPoints)
+    points      = models.ManyToManyField(JuryPoints, null=True, blank=True)
```
＊ ＊ ＊
Thank you for your work, you rock :+1:!
